### PR TITLE
Json keys

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

part of: #6215
fixes: #6214

First pass of adding QTT tests around the JSON key format, including adding tests for `DECIMAL` and `BOOLEAN` types, which is the first time they've been used as a key column type.

Commit includes a fix in `SerdeOptionsFactory` to ensure no key unwrapping is set if there is no key, plus refactored `SerdeOptionsFactory`, combining the two public methods.

### Testing done 

usual

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/json.json [hunk 1]
```diff
@@ -1,31 +1,426 @@
...
+        {"topic": "input_topic", "value": {"V": 1234567890}},
+        {"topic": "input_topic", "value": null}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": 1234567890},
+        {"topic": "OUTPUT", "value": null}
+      ]
+    },
+    {
+      "name": "DOUBLE - key",
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/json.json:154]: The PR description says this test is currently disabled. How so?

**Review by vcrfxia — APPROVED** [review]
LGTM though I think it's worth adding a few QTTs for converting between different key formats, both positive and negative such as validating what happens if I start with a JSON boolean key and try converting the key format to KAFKA.

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/json.json (hunk 1 above)
Comment (big-andy-coates) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/json.json:154]: Ah, I'll update the title.  I pushed another commit after creating the PR to enable the double testing.

**Comment by big-andy-coates** [comment]
> LGTM though I think it's worth adding a few QTTs for converting between different key formats, both positive and negative such as validating what happens if I start with a JSON boolean key and try converting the key format to KAFKA.

See https://github.com/confluentinc/ksql/pull/6345

**Pull request merged**